File: Medium_problems/ReversePolishNotation_150/Sol.py
class Solution:
    def reverse_polish_notation(self, tokens):
        stack = []
        for string in tokens:
            if string not in "*/-+":
                stack.append(int(string))
            else:
                if string == '+':
                    stack.append(stack.pop()+stack.pop())
                elif string=='-':
                    stack.append(-stack.pop()+stack.pop())
                elif string=='*':
                    stack.append(stack.pop()*stack.pop())
                else:
                    stack.append(int(1.0/stack.pop()*stack.pop()))
            
        return stack[0]

File: Medium_problems/ReversePolishNotation_150/Sol.java
package Medium_problems.ReversePolishNotation_150;

import java.util.Stack;

public class Sol {
    public static void main(String[] args) {
        System.out.println(ReversePolishNotation_150(new String[]{"4","13","5","/","+"}));
    }

    public static int ReversePolishNotation_150(String tokens[]){
        Stack<Integer> stack =  new Stack<>();

        for(String s : tokens)
            switch (s) {
                case "+":
                    stack.push(stack.pop()+stack.pop());
                    break;
                case "-":
                    stack.push(-stack.pop()+stack.pop());
                    break;
                case "*":
                    stack.push(stack.pop()*stack.pop());
                    break;
                case "/":  
                   stack.push(Double.valueOf(1.0/stack.pop()*stack.pop()).intValue());
                   break;
                default:
                    stack.push(Integer.valueOf(s));
                    break;
            }
        
        return stack.pop();
    }
}
